jgrapht-core/src/main/java/org/jgrapht/alg/matching/blossom/v5/KolmogorovWeightedPerfectMatching.java

```diff
@@ -157,22 +201,31 @@ public KolmogorovMinimumWeightPerfectMatching(Graph<V, E> graph, BlossomVOptions
     }
 
     /**
-     * Computes and returns a minimum weight perfect matching in the {@code graph}. See the class description
+     * Computes and returns a weighted perfect matching in the {@code graph}. See the class description
      * for the relative definitions and algorithm description.
      *
-     * @return the minimum weight perfect matching for the {@code graph}
+     * @return a weighted perfect matching for the {@code graph}
      */
     @Override
     public MatchingAlgorithm.Matching<V, E> getMatching() {
         if (matching == null) {
-            lazyComputeMinimumWeightPerfectMatching();
+            if (maximize) {
+                graph.edgeSet().forEach(e -> graph.setEdgeWeight(e, -graph.getEdgeWeight(e)));
```

Comment:
Let's not modify the user's graph. Instead, use:
`this.graph =  new AsWeightedGraph(graph, e -> graph.getEdgeWeight(e) * -1, true, false);`